dirsrvtests/tests/suites/export/export_test.py

```diff
@@ -0,0 +1,101 @@
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2020 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+
+
+import os
+import pytest
+import subprocess
+
+from lib389.topologies import topology_st as topo
+from lib389._constants import DEFAULT_SUFFIX, DEFAULT_BENAME
+from lib389.dbgen import dbgen_users
+from lib389.tasks import ImportTask
+from lib389.utils import *
+from lib389.idm.user import UserAccount, UserAccounts
+from lib389.idm.account import Accounts, Account
+from lib389.paths import Paths
+
+pytestmark = pytest.mark.tier1
+
+
+@pytest.mark.bz1806978
+@pytest.mark.ds51188
+@pytest.mark.skipif(ds_is_older("1.4.3"), reason="Not implemented")
+def test_db2ldif_with_non_accessible_ldif_file_path(topo):
+    """Export with db2ldif, giving a ldif file path which can't be accessed by the user (dirsrv by default)
+
+    :id: ca91eda7-27b1-4750-a013-531a63d3f5b0
+    :setup: Standalone Instance - entries imported in the db
+    :steps:
+        1. Stop the server
+        2. Launch db2ldif with an non accessible ldif file path 
+        3. Catch the reported error code
+        4. check the error reported in the errors log
+    :expected results:
+        1. Operation successful
+        2. Operation properly fails, without crashing
+        3. An error code different from 139 (segmentation fault) should be reported
+        4. 'ERR - bdb_db2ldif - db2ldif: userRoot: can't open file' should be reported
+    """
+    export_ldif = '/tmp/nonexistent/export.ldif'
+
+    log.info("Stopping the instance...")
+    topo.standalone.stop()
+
+    log.info("Performing an offline export to a non accessible ldif file path - should fail properly")
+    try:
+        subprocess.check_call(['db2ldif', '-Z', topo.standalone.serverid, '-s', DEFAULT_SUFFIX, '-a', export_ldif]) 
```

Comment:
I think you should use the topo db2ldif call instead. The reason is that if you are a developer (like myself) then I install into a prefix (/opt/dirsrv) and db2ldif won't be on my shell path, so subprocess here would fail. topo.db2ldif is aware of prefixs, and knows about the server id etc, so you should call `topo.standalone.db2ldif( ... ) ` instead.

Reply:
Or if that's not possible, at least use `os.path.join(self.ds_paths.sbin_dir, 'db2ldif')` to account for prefixs.